engine.py

```diff
@@ -755,6 +767,35 @@ def show_panel(self, panel_id, title, bundle, widget_class, *args, **kwargs):
         # resolved by looking at the stream of event and force triggering refreshes at the
         # right locations
         #
-        tk_maya.install_callbacks(panel_id, widget_id)
+        tk_maya.install_callbacks(maya_panel_name, widget_id)
 
         return widget_instance
+
+    def close_dialogs_and_panels(self):
+        """
+        Closes the dialogs and panels opened by the engine.
+        """
+
+        # Make a copy of the list of Tank dialogs that have been created by the engine and
```

Comment:
general thought: What happens if there is an error? I think it might make sense to try to catch any exceptions, report them as errors and then continue. This would mean that the method is *trying* to close all windows and if it fails to close a window, it continues rather than block further execution.

I can imagine strange reasons why things may fail - modality, perhaps strange docking related reasons, external processes having closed the window or altered it in strange ways etc.